docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
+Note that proxying of application traffic has to be configured on the application side.
 
-The cost is a number between 0 and 255, with zero (the default) implying free, and less preferred ports being assigned a higher cost. Multiple ports can be assigned the same cost, in which case once failover has happened to cost N, then all uplink ports with cost N will be used for load spreading of the management traffic.
+## Network Time Protocol (NTP)
 
-Furthermore, one can set [network.download.max.cost](CONFIG-PROPERTIES.md) to a number N if it is acceptable for EVE to perform (potentially large) downloads of content and images when having failed over to uplink ports with cost N.
+EVE always tries to keep its clock synchronized by embedding a NTP Client that synchronizes
+the system's clock since the boot time.
+This is especially important on the very first boot to ensure that all device certificates
+are generated with valid timestamps. EVE will first try to use NTP servers either received
+from DHCP servers or statically configured in the network override configuration used for
+bootstrapping (see the next section for more info on the network bootstrapping process).
+If no NTP server is provided by DHCP or config, EVE will try to use `pool.ntp.org` by default.
 
-The cost is new and will replace the free/freeUplink way to specify two levels of cost (free or paid). For compatibility reasons EVE will look at freeUplink for the SystemAdapter and PhysicalIO so that the free/paid distinction still works until the cost parameter is used by all the controller.
+For more info on clock synchronization, refer to [CLOCK-SYNCHRONIZATION.md](CLOCK-SYNCHRONIZATION.md).
 
 ## Sources of configuration
 
-There are several sources from which nim gets the potential port configurations. Those all use the ```DevicePortConfig``` type. There are examples of such configurations in [legacy EVE configuration](CONFIG.md)
+There are several sources from which nim gets the potential port configurations.
+Those all use the ```DevicePortConfig``` type (Go struct marshalled with JSON).
+There are examples of such configurations in [legacy EVE configuration](CONFIG.md)
 
 ### Fresh install
 
-When EVE is installed using a generic EVE installer without device bootstrap configuration included (see [CONFIG.md](./CONFIG.md), section "Bootstrap configuration"), it only has the last-resort configuration available (see "Last resort" section below). This is sufficient to connect to the controller if
+When EVE is installed using a generic EVE installer without device bootstrap configuration
+included (see [CONFIG.md](./CONFIG.md), section "Bootstrap configuration"), it only has
+the last-resort configuration available (see "Last resort" section below). This is sufficient
+to connect to the controller if
 
-- DHCP is enabled on one of the Ethernet ports
+- DHCP is enabled on one of the uplink Ethernet ports
```

Comment:
Nit, but why did you add "uplink" here? If there is no bootstrap configuration then all Ethernet ports are tried.

Reply:
These are the requirements for working connectivity with the cloud when lastresort is used. So it is not about what EVE will try but what must be provided.

Reply:
Oh - I misread the sentence above it.